.github/workflows/build-docker.yml

```diff
@@ -0,0 +1,56 @@
+name: Reusable Docker build and push workflow
+
+on:
+  workflow_call:
+    inputs:
+      registry:
+        required: true
+        type: string
+      username:
+        required: true
+        type: string
+      app_name:
+        required: true
+        type: string
+      dockerfile:
+        required: true
+        type: string
+      push:
+        required: true
+        type: boolean
+    secrets:
+      DOCKER_PASSWORD:
+        required: true
+
+jobs:
+  docker-build:
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Docker Login
+      uses: docker/login-action@v3
+      with:
+        registry: ${{ inputs.registry }}
+        username: ${{ inputs.username }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Get current date
+      id: date
+      run: echo "date=$(date +'%Y-%m-%d')" >> $GITHUB_OUTPUT
+
+    - name: Build and push the Docker image
```

Comment:
Just wondering if we should rename this? Sometimes we don't push now, might be confusing

Reply:
Sure. How's "Reusable Docker build and optional push workflow"?

Reply:
I think we can omit "Reusable", it gets outputted in the workflow run which would look a bit odd and we have it in the workflow name already.

Maybe "Build Docker Image (and push if on main branch)". Still not great but does what it says on the tin 🙈

Reply:
I'm calling it Build docker image and optionally push, since this file says nothing about main branch - that config is set by the workflow that uses it. That should satisfy your requirement of making it clear we don't always push - and it doesn't say "reusable". Sound good?